--- /usr/portage/www-client/chromium/chromium-65.0.3298.3.ebuild	2018-01-12 22:10:09.000000000 +0200
+++ chromium/chromium-65.0.3298.3.ebuild	2018-01-15 13:34:28.177196008 +0200
@@ -17,7 +17,7 @@
 LICENSE="BSD"
 SLOT="0"
 KEYWORDS="~amd64 ~arm ~arm64 ~x86"
-IUSE="component-build cups gnome-keyring +hangouts jumbo-build kerberos neon pic +proprietary-codecs pulseaudio selinux +suid +system-ffmpeg +system-icu +system-libvpx +tcmalloc widevine"
+IUSE="component-build cups gnome-keyring +hangouts jumbo-build kerberos neon pic +proprietary-codecs pulseaudio selinux +suid +system-ffmpeg +system-icu +system-libvpx +tcmalloc vaapi widevine"
 RESTRICT="!system-ffmpeg? ( proprietary-codecs? ( bindist ) )"
 
 COMMON_DEPEND="
@@ -85,6 +85,7 @@
 	virtual/ttf-fonts
 	selinux? ( sec-policy/selinux-chromium )
 	tcmalloc? ( !<x11-drivers/nvidia-drivers-331.20 )
+	vaapi? ( x11-libs/libva )
 	widevine? ( www-plugins/chrome-binary-plugins[widevine(-)] )
 "
 # dev-vcs/git - https://bugs.gentoo.org/593476
@@ -150,6 +151,7 @@
 	"${FILESDIR}/chromium-memcpy-r0.patch"
 	"${FILESDIR}/chromium-clang-r2.patch"
 	"${FILESDIR}/chromium-gcc-u2f.patch"
+	"${FILESDIR}/chromium-intel-vaapi_r16.diff.patch"
 )
 
 pre_build_checks() {
@@ -436,6 +438,9 @@
 	myconf_gn+=" use_kerberos=$(usex kerberos true false)"
 	myconf_gn+=" use_pulseaudio=$(usex pulseaudio true false)"
 
+	# USE-enable VAAPI
+	myconf_gn+=" use_vaapi=$(usex vaapi true false)"
+
 	# TODO: link_pulseaudio=true for GN.
 
 	myconf_gn+=" fieldtrial_testing_like_official_build=true"
